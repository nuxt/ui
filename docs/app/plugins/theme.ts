export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
    const appConfig = useAppConfig()

    if (import.meta.client) {
      function updateColor(type: 'primary' | 'neutral') {
        const color = localStorage.getItem(`nuxt-ui-${type}`)
        if (color) {
          appConfig.ui.colors[type] = color
        }
      }

      function updateRadius() {
        const radius = localStorage.getItem('nuxt-ui-radius')
        if (radius) {
          appConfig.theme.radius = Number.parseFloat(radius)
        }
      }

      updateColor('primary')
      updateColor('neutral')
      updateRadius()
    }

    if (import.meta.server) {
      useHead({
        script: [{
          innerHTML: `
            let html = document.querySelector('style#nuxt-ui-colors').innerHTML;

            if (localStorage.getItem('nuxt-ui-primary')) {
              const primaryColor = localStorage.getItem('nuxt-ui-primary');
              html = html.replace(
                /(--ui-color-primary-\\d{2,3}:\\s*var\\()--color-${appConfig.ui.colors.primary}-(\\d{2,3}\\))/g,
                \`$1--color-\${primaryColor}-$2\`
              );
            }
            if (localStorage.getItem('nuxt-ui-neutral')) {
              const neutralColor = localStorage.getItem('nuxt-ui-neutral');
              html = html.replace(
                /(--ui-color-neutral-\\d{2,3}:\\s*var\\()--color-${appConfig.ui.colors.neutral}-(\\d{2,3}\\))/g,
                \`$1--color-\${neutralColor}-$2\`
              );
            }

            document.querySelector('style#nuxt-ui-colors').innerHTML = html;
            `.replace(/\s+/g, ' '),
          type: 'text/javascript',
          tagPriority: -1
        }, {
          innerHTML: `
            if (localStorage.getItem('nuxt-ui-radius')) {
              document.querySelector('style#nuxt-ui-radius').innerHTML = ':root { --ui-radius: ' + localStorage.getItem('nuxt-ui-radius') + 'rem; }';
            }
          `.replace(/\s+/g, ' '),
          type: 'text/javascript',
          tagPriority: -1
        }]
      })
    }
  }
})
