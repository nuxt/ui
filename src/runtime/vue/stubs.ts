import { ref } from 'vue'
import type { Ref, Plugin as VuePlugin } from 'vue'

import appConfig from '#build/app.config'
import type { NuxtApp } from '#app'

export { useHead } from '@unhead/vue'
export { useRoute, useRouter } from 'vue-router'

export const useAppConfig = () => appConfig

const state: Record<string, any> = {}

export const useState = <T>(key: string, init: () => T): Ref<T> => {
  if (state[key]) {
    return state[key] as Ref<T>
  }
  const value = ref(init())
  state[key] = value
  return value as Ref<T>
}

export function useNuxtApp() {
  return {
    isHydrating: true,
    payload: { serverRendered: false }
  }
}

export function defineNuxtPlugin(plugin: (nuxtApp: NuxtApp) => void) {
  return {
    install(app) {
      plugin({ vueApp: app } as NuxtApp)
    }
  } satisfies VuePlugin
}
