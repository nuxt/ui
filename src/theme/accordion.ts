export default {
  slots: {
    root: 'w-full',
    item: 'border-b border-[var(--ui-border)] last:border-b-0',
    header: 'flex',
    trigger: 'group flex-1 flex items-center gap-1.5 font-medium text-sm py-3.5 focus-visible:outline-[var(--ui-primary)] min-w-0',
    content: 'data-[state=open]:animate-[accordion-down_200ms_ease-out] data-[state=closed]:animate-[accordion-up_200ms_ease-out] overflow-hidden focus:outline-none',
    body: 'text-sm pb-3.5',
    leadingIcon: 'shrink-0 size-5',
    trailingIcon: 'shrink-0 size-5 ms-auto group-data-[state=open]:rotate-180 transition-transform duration-200',
    label: 'text-start break-words'
  },
  variants: {
    disabled: {
      true: {
        trigger: 'cursor-not-allowed opacity-75'
      }
    }
  }
}
