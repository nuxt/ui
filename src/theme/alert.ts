import type { ModuleOptions } from '../module'

export default (options: Required<ModuleOptions>) => ({
  slots: {
    root: 'relative overflow-hidden w-full rounded-[calc(var(--ui-radius)*2)] p-4 flex gap-2.5',
    wrapper: 'min-w-0 flex-1 flex flex-col gap-1',
    title: 'text-sm font-medium',
    description: 'text-sm opacity-90',
    icon: 'shrink-0 size-5',
    avatar: 'shrink-0',
    avatarSize: '2xl',
    actions: 'flex gap-1.5 shrink-0',
    close: 'p-0.5'
  },
  variants: {
    color: {
      ...Object.fromEntries((options.theme.colors || []).map((color: string) => [color, ''])),
      neutral: ''
    },
    variant: {
      solid: '',
      outline: '',
      soft: '',
      subtle: ''
    },
    multiline: {
      true: {
        root: 'items-start',
        actions: 'items-start mt-1'
      },
      false: {
        root: 'items-center',
        actions: 'items-center'
      }
    }
  },
  compoundVariants: [...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'solid',
    class: {
      root: `bg-[var(--ui-${color})] text-[var(--ui-bg)]`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'outline',
    class: {
      root: `text-[var(--ui-${color})] ring ring-inset ring-[var(--ui-${color})]/25`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'soft',
    class: {
      root: `bg-[var(--ui-${color})]/10 text-[var(--ui-${color})]`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'subtle',
    class: {
      root: `bg-[var(--ui-${color})]/10 text-[var(--ui-${color})] ring ring-inset ring-[var(--ui-${color})]/25`
    }
  })), {
    color: 'neutral',
    variant: 'solid',
    class: {
      root: 'text-[var(--ui-bg)] bg-[var(--ui-bg-inverted)]'
    }
  }, {
    color: 'neutral',
    variant: 'outline',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg)] ring ring-inset ring-[var(--ui-border)]'
    }
  }, {
    color: 'neutral',
    variant: 'soft',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg-elevated)]/50'
    }
  }, {
    color: 'neutral',
    variant: 'subtle',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg-elevated)]/50 ring ring-inset ring-[var(--ui-border-accented)]'
    }
  }],
  defaultVariants: {
    color: 'primary',
    variant: 'solid'
  }
})
