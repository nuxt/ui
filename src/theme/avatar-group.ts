export default {
  slots: {
    root: 'inline-flex flex-row-reverse justify-end',
    base: 'relative rounded-full ring-[var(--ui-bg)] first:me-0'
  },
  variants: {
    size: {
      '3xs': {
        base: 'ring -me-0.5'
      },
      '2xs': {
        base: 'ring -me-0.5'
      },
      'xs': {
        base: 'ring -me-0.5'
      },
      'sm': {
        base: 'ring-2 -me-1.5'
      },
      'md': {
        base: 'ring-2 -me-1.5'
      },
      'lg': {
        base: 'ring-2 -me-1.5'
      },
      'xl': {
        base: 'ring-3 -me-2'
      },
      '2xl': {
        base: 'ring-3 -me-2'
      },
      '3xl': {
        base: 'ring-3 -me-2'
      }
    }
  },
  defaultVariants: {
    size: 'md'
  }
}
