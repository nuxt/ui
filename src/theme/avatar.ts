export default {
  slots: {
    root: 'inline-flex items-center justify-center shrink-0 select-none overflow-hidden rounded-full align-middle bg-[var(--ui-bg-elevated)]',
    image: 'h-full w-full rounded-[inherit] object-cover',
    fallback: 'font-medium leading-none text-[var(--ui-text-muted)] truncate',
    icon: 'text-[var(--ui-text-muted)] shrink-0'
  },
  variants: {
    size: {
      '3xs': {
        root: 'size-4 text-[8px]'
      },
      '2xs': {
        root: 'size-5 text-[10px]'
      },
      'xs': {
        root: 'size-6 text-xs'
      },
      'sm': {
        root: 'size-7 text-sm'
      },
      'md': {
        root: 'size-8 text-base'
      },
      'lg': {
        root: 'size-9 text-lg'
      },
      'xl': {
        root: 'size-10 text-xl'
      },
      '2xl': {
        root: 'size-11 text-[22px]'
      },
      '3xl': {
        root: 'size-12 text-2xl'
      }
    }
  },
  defaultVariants: {
    size: 'md'
  }
}
