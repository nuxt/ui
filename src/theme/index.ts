export { default as accordion } from './accordion'
export { default as alert } from './alert'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as breadcrumb } from './breadcrumb'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as calendar } from './calendar'
export { default as card } from './card'
export { default as carousel } from './carousel'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as collapsible } from './collapsible'
export { default as colorPicker } from './color-picker'
export { default as commandPalette } from './command-palette'
export { default as container } from './container'
export { default as contextMenu } from './context-menu'
export { default as drawer } from './drawer'
export { default as dropdownMenu } from './dropdown-menu'
export { default as form } from './form'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as inputMenu } from './input-menu'
export { default as inputNumber } from './input-number'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as modal } from './modal'
export { default as navigationMenu } from './navigation-menu'
export { default as pagination } from './pagination'
export { default as pinInput } from './pin-input'
export { default as popover } from './popover'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as select } from './select'
export { default as selectMenu } from './select-menu'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as slideover } from './slideover'
export { default as slider } from './slider'
export { default as stepper } from './stepper'
export { default as switch } from './switch'
export { default as table } from './table'
export { default as tabs } from './tabs'
export { default as textarea } from './textarea'
export { default as toast } from './toast'
export { default as toaster } from './toaster'
export { default as tooltip } from './tooltip'
