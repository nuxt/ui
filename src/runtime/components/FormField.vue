<script lang="ts">
import { tv, type VariantProps } from 'tailwind-variants'
import type { AppConfig } from '@nuxt/schema'
import _appConfig from '#build/app.config'
import theme from '#build/ui/form-field'
import { extendDevtoolsMeta } from '../composables/extendDevtoolsMeta'

const appConfig = _appConfig as AppConfig & { ui: { formField: Partial<typeof theme> } }

const formField = tv({ extend: tv(theme), ...(appConfig.ui?.formField || {}) })

type FormFieldVariants = VariantProps<typeof formField>

export interface FormFieldProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  /** The name of the FormField. Also used to match form errors. */
  name?: string
  /** A regular expression to match form error names. */
  errorPattern?: RegExp
  label?: string
  description?: string
  help?: string
  error?: string | boolean
  hint?: string
  size?: FormFieldVariants['size']
  required?: boolean
  eagerValidation?: boolean
  validateOnInputDelay?: number
  class?: any
  ui?: Partial<typeof formField.slots>
}

export interface FormFieldSlots {
  label(props: { label?: string }): any
  hint(props: { hint?: string }): any
  description(props: { description?: string }): any
  help(props: { help?: string }): any
  error(props: { error?: string | boolean }): any
  default(props: { error?: string | boolean }): any
}

extendDevtoolsMeta({ example: 'FormFieldExample', defaultProps: { label: 'Label' } })
</script>

<script setup lang="ts">
import { computed, ref, inject, provide, type Ref, useId } from 'vue'
import { Primitive, Label } from 'reka-ui'
import { formFieldInjectionKey, inputIdInjectionKey } from '../composables/useFormField'
import type { FormError, FormFieldInjectedOptions } from '../types/form'

const props = defineProps<FormFieldProps>()
const slots = defineSlots<FormFieldSlots>()

const ui = computed(() => formField({
  size: props.size,
  required: props.required
}))

const formErrors = inject<Ref<FormError[]> | null>('form-errors', null)

const error = computed(() => props.error || formErrors?.value?.find(error => error.name === props.name || (props.errorPattern && error.name.match(props.errorPattern)))?.message)

const id = ref(useId())

provide(inputIdInjectionKey, id)

provide(formFieldInjectionKey, computed(() => ({
  error: error.value,
  name: props.name,
  size: props.size,
  eagerValidation: props.eagerValidation,
  validateOnInputDelay: props.validateOnInputDelay,
  errorPattern: props.errorPattern
}) as FormFieldInjectedOptions<FormFieldProps>))
</script>

<template>
  <Primitive :as="as" :class="ui.root({ class: [props.class, props.ui?.root] })">
    <div :class="ui.wrapper({ class: props.ui?.wrapper })">
      <div v-if="label || !!slots.label" :class="ui.labelWrapper({ class: props.ui?.labelWrapper })">
        <Label :for="id" :class="ui.label({ class: props.ui?.label })">
          <slot name="label" :label="label">
            {{ label }}
          </slot>
        </Label>
        <span v-if="hint || !!slots.hint" :class="ui.hint({ class: props.ui?.hint })">
          <slot name="hint" :hint="hint">
            {{ hint }}
          </slot>
        </span>
      </div>

      <p v-if="description || !!slots.description" :class="ui.description({ class: props.ui?.description })">
        <slot name="description" :description="description">
          {{ description }}
        </slot>
      </p>
    </div>

    <div :class="[(label || !!slots.label || description || !!slots.description) && ui.container({ class: props.ui?.container })]">
      <slot :error="error" />

      <p v-if="(typeof error === 'string' && error) || !!slots.error" :class="ui.error({ class: props.ui?.error })">
        <slot name="error" :error="error">
          {{ error }}
        </slot>
      </p>
      <p v-else-if="help || !!slots.help" :class="ui.help({ class: props.ui?.help })">
        <slot name="help" :help="help">
          {{ help }}
        </slot>
      </p>
    </div>
  </Primitive>
</template>
