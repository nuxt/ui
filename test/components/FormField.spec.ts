import { defineComponent } from 'vue'
import { describe, it, expect } from 'vitest'
import FormField, { type FormFieldProps, type FormFieldSlots } from '../../src/runtime/components/FormField.vue'
import ComponentRender from '../component-render'
import theme from '#build/ui/form-field'

// A wrapper component is needed here because of a conflict with the error prop / expose.
// See: https://github.com/nuxt/test-utils/issues/684
const FormFieldWrapper = defineComponent({
  components: {
    UFormField: FormField
  },
  template: `<UFormField>
  <template v-for="(_, name) in $slots" #[name]="slotData">
    <slot :name="name" v-bind="slotData" />
  </template>
</UFormField>`
})

describe('FormField', () => {
  const sizes = Object.keys(theme.variants.size) as any

  it.each([
    // Props
    ['with label and description', { props: { label: 'Username', description: 'Enter your username' } }],
    ['with required', { props: { label: 'Username', required: true } }],
    ['with help', { props: { help: 'Username must be unique' } }],
    ['with error', { props: { error: 'Username is already taken' } }],
    ['with hint', { props: { hint: 'Use letters, numbers, and special characters' } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { label: 'Username', description: 'Enter your username', size } }]),
    ['with as', { props: { as: 'section' } }],
    ['with class', { props: { class: 'relative' } }],
    ['with ui', { props: { ui: { label: 'text-[var(--ui-text-highlighted)]' } } }],
    // Slots
    ['with default slot', { slots: { default: () => 'Default slot' } }],
    ['with label slot', { slots: { label: () => 'Label slot' } }],
    ['with description slot', { slots: { description: () => 'Description slot' } }],
    ['with error slot', { slots: { error: () => 'Error slot' } }],
    ['with hint slot', { slots: { hint: () => 'Hint slot' } }],
    ['with help slot', { slots: { help: () => 'Help slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: FormFieldProps, slots?: Partial<FormFieldSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, FormFieldWrapper)
    expect(html).toMatchSnapshot()
  })
})
