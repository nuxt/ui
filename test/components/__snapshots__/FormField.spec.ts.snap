// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormField > renders with class correctly 1`] = `
"<div class="text-sm relative">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with default slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">Default slot
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with description slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <p class="text-gray-500 dark:text-gray-400">Description slot</p>
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with error correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <p class="mt-2 text-error-500 dark:text-error-400">Username is already taken</p>
  </div>
</div>"
`;

exports[`FormField > renders with error slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <p class="mt-2 text-error-500 dark:text-error-400">Error slot</p>
  </div>
</div>"
`;

exports[`FormField > renders with help correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <p class="mt-2 text-gray-500 dark:text-gray-400">Username must be unique</p>
  </div>
</div>"
`;

exports[`FormField > renders with help slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <p class="mt-2 text-gray-500 dark:text-gray-400">Help slot</p>
  </div>
</div>"
`;

exports[`FormField > renders with hint correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with hint slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with label and description correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with label slot correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Label slot</label>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with required correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200 after:content-['*'] after:ms-0.5 after:text-error-500 dark:after:text-error-400">Username</label>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with size lg correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with size md correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with size sm correctly 1`] = `
"<div class="text-xs">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with size xl correctly 1`] = `
"<div class="text-base">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with size xs correctly 1`] = `
"<div class="text-xs">
  <div class="">
    <div class="flex content-center items-center justify-between"><label for="v-0-0" class="block font-medium text-gray-700 dark:text-gray-200">Username</label>
      <!--v-if-->
    </div>
    <p class="text-gray-500 dark:text-gray-400">Enter your username</p>
  </div>
  <div class="mt-1 relative">
    <!--v-if-->
  </div>
</div>"
`;

exports[`FormField > renders with ui correctly 1`] = `
"<div class="text-sm">
  <div class="">
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="">
    <!--v-if-->
  </div>
</div>"
`;
